pages/find/index.js

```diff
@@ -2,44 +2,43 @@ import React, { useEffect } from "react"
 import { Input } from "../../components/ui/form";
 import Button from "../../components/ui/Button";
 import Router from "next/router";
-import {Alert} from "react-bootstrap";
+import { Alert } from "react-bootstrap";
 
 const Find = () => {
     const [token, setToken] = React.useState("");
-    const [to, setTo] = React.useState("");
     const [loading, setLoading] = React.useState(false);
-    const [notify, setNotify] = React.useState({msg: "", type: ""});
+    const [notify, setNotify] = React.useState({ msg: "", type: "" });
     const [show, setShow] = React.useState(true);
 
-    const onSubmit = (e) =>{
-        e.preventDefault();
-        fetch(`${process.env.NEXT_PUBLIC_BC_HOST}/${token}`)
-        .then(res =>  res.json())
-        .then(data => {
-            if(token.length < 1 || data.status_code === 404 ) {
-            setNotify({msg:"Invalid Token or Certificate not found", type:"error"})
-            } else{
-                setLoading(true);
-                Router.push(`/${to}/${token}`)}
-            }
-        )
-        .catch(err => err)
+    const onSubmit = async (previewFormat) => {
+        try {
+            if(!token) return;
```

Comment:
You should set loading as true once you start the loading process which is after this validation